Add memory_format support to `zeros`, `ones`, `full`

VitalyFedyunin opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #31644 [WIP] Fix cudnn channels_last descriptio problem
* **#31131 Add memory_format support to `zeros`, `ones`, `full`**

Differential Revision: [D18959228](https://our.internmc.facebook.com/intern/diff/D18959228)

VitalyFedyunin committed:
Add memory_format support to `zeros`, `ones`, `full` (still need tests)

[ghstack-poisoned]

VitalyFedyunin committed:
no-tags channels last solution on "Add memory_format support to `zeros`, `ones`, `full` (still need tests)"




[ghstack-poisoned]

VitalyFedyunin committed:
rebase to master on "Add memory_format support to `zeros`, `ones`, `full` (still need tests)"




[ghstack-poisoned]

kostmo commented:
## CircleCI build failures summary
As of commit 69f22b6:
* **1/1** failures introduced in this PR



## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=69f22b6020afa55ebf836f581dc72341789da3ed).

### 1 new failure recognized by patterns
The following build failures don't appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3924850) pytorch-linux-xenial-py3-clang5-android-ndk-r19c-gradle-build (1/1)
**Step:** "pytorch android gradle build" ([details](https://dr.pytorch.org/build-details.html?build_id=73319146))
```
sys	0m0.159s
++ docker run --cap-add=SYS_PTRACE --security-opt seccomp=unconfined -t -d -w /var/lib/jenkins 308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-xenial-py3-clang5-android-ndk-r19c:405-69f22b6020afa55ebf836f581dc72341789da3ed-android-x86_32
+ export id_x86_32=4d48574c8408f05ddc84072293aaebb70dd875f86ca01d5b0c2ce3f6af29c14d
+ id_x86_32=4d48574c8408f05ddc84072293aaebb70dd875f86ca01d5b0c2ce3f6af29c14d
+ export 'COMMAND=((echo "source ./workspace/env" && echo "sudo chown -R jenkins workspace") | docker exec -u jenkins -i "$id_x86_32" bash) 2>&1'
+ COMMAND='((echo "source ./workspace/env" && echo "sudo chown -R jenkins workspace") | docker exec -u jenkins -i "$id_x86_32" bash) 2>&1'
+ echo '((echo' '"source' './workspace/env"' '&&' echo '"sudo' chown -R jenkins 'workspace")' '|' docker exec -u jenkins -i '"$id_x86_32"' 'bash)' '2>&1'
+ unbuffer bash ./command.sh
+ ts
+ docker pull 308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-xenial-py3-clang5-android-ndk-r19c:405-69f22b6020afa55ebf836f581dc72341789da3ed-android-arm-v7a
Error response from daemon: manifest for 308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-xenial-py3-clang5-android-ndk-r19c:405-69f22b6020afa55ebf836f581dc72341789da3ed-android-arm-v7a not found
```


---
This comment was automatically generated by [Dr. CI](https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment).
Follow [this link to opt-out](https://dr.pytorch.org/admin/comments-opt-out.html) of these comments for your Pull Requests.


Please report bugs/suggestions on the [GitHub issue tracker](https://github.com/kostmo/circleci-failure-tracker/issues).

VitalyFedyunin committed:
Fix flake8 on "Add memory_format support to `zeros`, `ones`, `full` (still need tests)"


Differential Revision: [D18959228](https://our.internmc.facebook.com/intern/diff/D18959228)

[ghstack-poisoned]

VitalyFedyunin committed:
Fix flake8 on "Add memory_format support to `zeros`, `ones`, `full` (still need tests)"


Differential Revision: [D18959228](https://our.internmc.facebook.com/intern/diff/D18959228)

[ghstack-poisoned]

VitalyFedyunin committed:
Add tests on "Add memory_format support to `zeros`, `ones`, `full` (still need tests)"


Differential Revision: [D18959228](https://our.internmc.facebook.com/intern/diff/D18959228)

[ghstack-poisoned]

kostmo commented:
## CircleCI build failures summary
As of commit ca1e0a4:
* **1/1** failures introduced in this PR



## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=ca1e0a4d8c86578c98b56ff213492369da5186be).

### 1 new failure recognized by patterns
The following build failures don't appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3965501) pytorch_windows_build (1/1)
**Step:** "Build" ([details](https://dr.pytorch.org/build-details.html?build_id=75227349))
<details>
<summary>
<code>CMake Error at torch_cuda_generated_THCTensorIndex.cu.obj.Release.cmake:281 (message):</code>
</summary>

```
C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v10.1/include\thrust/detail/advance.inl(33): error C2923: 'std::_Select<__formal>::_Apply': '_1' is not a valid template type argument for parameter '<unnamed-symbol>'
C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v10.1/include\thrust/detail/advance.inl(33): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v10.1/include\thrust/detail/advance.inl(33): error C2144: syntax error: 'unknown-type' should be preceded by ')'
C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v10.1/include\thrust/detail/advance.inl(33): error C2144: syntax error: 'unknown-type' should be preceded by ';'
C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v10.1/include\thrust/detail/advance.inl(33): error C2238: unexpected token(s) preceding ';'
C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v10.1/include\thrust/detail/advance.inl(33): error C2059: syntax error: ')'
C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v10.1/include\thrust/detail/advance.inl(33): error C2988: unrecognizable template declaration/definition
C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v10.1/include\thrust/detail/advance.inl(33): error C2059: syntax error: '<end Parse>'
-- Removing C:/Users/circleci/project/build/caffe2/CMakeFiles/torch_cuda.dir/__/aten/src/THC/./torch_cuda_generated_THCTensorIndex.cu.obj
"C:/Program Files/CMake/bin/cmake.exe" -E remove C:/Users/circleci/project/build/caffe2/CMakeFiles/torch_cuda.dir/__/aten/src/THC/./torch_cuda_generated_THCTensorIndex.cu.obj
CMake Error at torch_cuda_generated_THCTensorIndex.cu.obj.Release.cmake:281 (message):
  Error generating file
  C:/Users/circleci/project/build/caffe2/CMakeFiles/torch_cuda.dir/__/aten/src/THC/./torch_cuda_generated_THCTensorIndex.cu.obj


RING=C:/Users/circleci/project/build/caffe2/CMakeFiles/torch_cuda.dir/__/aten/src/THC/generated/./torch_cuda_generated_THCTensorMathReduceShort.cu.obj -D generated_cubin_file:STRING=C:/Users/circleci/project/build/caffe2/CMakeFiles/torch_cuda.dir/__/aten/src/THC/generated/./torch_cuda_generated_THCTensorMathReduceShort.cu.obj.cubin.txt -P C:/Users/circleci/project/build/caffe2/CMakeFiles/torch_cuda.dir/__/aten/src/THC/generated/torch_cuda_generated_THCTensorMathReduceShort.cu.obj.Release.cmake" 
-- Removing C:/Users/circleci/project/build/caffe2/CMakeFiles/torch_cuda.dir/__/aten/src/THC/generated/./torch_cuda_generated_THCTensorMathReduceShort.cu.obj
"C:/Program Files/CMake/bin/cmake.exe" -E remove C:/Users/circleci/project/build/caffe2/CMakeFiles/torch_cuda.dir/__/aten/src/THC/generated/./torch_cuda_generated_THCTensorMathReduceShort.cu.obj
-- Generating dependency file: C:/Users/circleci/project/build/caffe2/CMakeFiles/torch_cuda.dir/__/aten/src/THC/generated/torch_cuda_generated_THCTensorMathReduceShort.cu.obj.NVCC-depend
C:/Users/circleci/project/aten/src/THCUNN -IC:/Users/circleci/project/aten/src/ATen/cuda -IC:/Users/circleci/project/build/caffe2/aten/src -IC:/Users/circleci/project/aten/../third_party/catch/single_include -IC:/Users/circleci/project/aten/src/ATen/.. -IC:/Users/circleci/project/build/caffe2/aten/src/ATen -IC:/Users/circleci/project/c10/cuda/../.. -IC:/Users/circleci/project/c10/../ -IC:/Users/circleci/project/caffe2/../torch/csrc/api -IC:/Users/circleci/project/caffe2/../torch/csrc/api/include
THCTensorMathReduceShort.cu 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

VitalyFedyunin committed:
experiemnts on "Add memory_format support to `zeros`, `ones`, `full`"


Differential Revision: [D18959228](https://our.internmc.facebook.com/intern/diff/D18959228)

[ghstack-poisoned]

VitalyFedyunin committed:
experiemnts on "Add memory_format support to `zeros`, `ones`, `full`"


Differential Revision: [D18959228](https://our.internmc.facebook.com/intern/diff/D18959228)

[ghstack-poisoned]
